drake/systems/plants/@Manipulator/Manipulator.m

```diff
@@ -85,24 +86,25 @@
         %   vdot = H\tau
         % but I already have and use Hinv, so use it again here
 
-        xdot = [vToqdot(obj,q)*v; vdot];
+        kinsol = obj.doKinematics(q);
+        xdot = [vToqdot(obj,kinsol)*v; vdot];
```

Comment:
It could. Not sure if that's a change we should be making now though. Optimizing the matlab side of things should be the least of our priorities in my opinion.